### PR TITLE
Mod restructuring

### LINKED ISSUE #297: Fix the gradle configurations to use artifacts

### PR DESCRIPTION
**Pull request by SigmundGranaas**
Restructuring the mod to make it easier to perform project builds for the complete version of the mods. Also makes use of the new multiproject optimizations, and gets rid of some hacks I did not like. This will also make publishing artifacts and managing dependencies a lot easier.

Also restructures the repository based on dependencies, and purpose. Content modules will be in the content directory, and Fabric modules are in the fabric module.

This will also make Forge support easier.

Also gets rid of the BETA classifier for built jars. Also uses the standard version format: 0.10.8+1.19.2

closes #297

### CODE REVIEW

**Comment by SigmundGranaas** [comment]
@Steveplays28 What do you think about structuring the project like this?

**Comment by Steveplays28** [comment]
Super nice work. I like the new folder structure and versioning. Following [semantic versioning](https://semver.org/) is a good idea indeed.
I'll review it more thoroughly throughout the day.

File: .gitignore [hunk 1]
```diff
@@ -118,34 +118,36 @@ output/
...
+core/logs/
+core/run/
+core/build/
+
+logs/
+run/
 /forgero-forge/run/
 /dest/
+
+fabric/logs/
```
Comment (Steveplays28) [inline .gitignore]: Couldn't we just use `*/logs/`, `*/run/`, and `*/build/` here, instead of the individual folders?

**Review by Steveplays28 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (SigmundGranaas) [reply .gitignore]: Yeah that's a much better solution

Comment (Steveplays28) [reply .gitignore]: Alrighty! That's the only thing I could see that would be nice to change.
PR looks super good for the rest, definitely a very welcome change to the folder and Gradle structure of Forgero. Nice work!

Comment (SigmundGranaas) [reply .gitignore]: Thanks! I'll merge it soon, so this will be the standard moving forward from 0.10.8

Comment (Steveplays28) [reply .gitignore]: Amazing!

**Pull request merged**